### ISSUE TITLE
[bug] WindowBuilder's .maximized not working on Windows 10

### ISSUE THREAD

**Issue by matthme** [tauri-apps/tauri#7987]
### Describe the bug

Using the WindowBuilder's [.maximized()](https://docs.rs/tauri/latest/tauri/window/struct.WindowBuilder.html#method.maximized) opens the window as a tiny window on Windows while it works fine on Linux (and acts weird on macOS, separate issue).

### Reproduction

```
WindowBuilder::new(
    app_handle,
    "main", 
    tauri::WindowUrl::App("index.html".into())
    ).center()
    .maximized(true)
    .build()
    .unwrap()
```

### Expected behavior

Window opens up maximized.

### Platform and versions

```text
Build environment (github CI):

[✘] Environment
    - OS: Windows 10.0.17763 X64
    ✘ Webview2: not installed!
      Visit https://developer.microsoft.com/en-us/microsoft-edge/webview2/
    ✔ MSVC: Visual Studio Enterprise 2019
    ✔ rustc: 1.73.0 (cc66ad468 2023-10-03)
    ✔ Cargo: 1.73.0 (9c4383fb5 2023-08-26)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (directory override for 'D:\a\wordcondenser-standalone\wordcondenser-standalone')
    - node: 16.20.2
    - yarn: 1.22.19
    - npm: 8.19.4

[-] Packages
    - tauri [RUST]: 1.5.1
    - tauri-build [RUST]: 1.5.0
    - wry [RUST]: 0.24.4
    - tao [RUST]: 0.16.4
    - tauri-cli [RUST]: not installed!
    - @tauri-apps/api [NPM]: not installed!
    - @tauri-apps/cli [NPM]: 1.5.1 (outdated, latest: 1.5.2)

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist
    - devPath: ../dist
```
The bug then occurs on a Windows 10 PC.
```


### Stack trace

_No response_

### Additional context

_No response_

**Comment by amrbashir** [comment]
I can't reproduce this at all, please provide a minimal reproducible example/repo.

**Comment by matthme** [comment]
Okay, I figured out what the problem was. I called `.center()` as well before calling `.maximized(true)`. I guess it's debatable whether that's a bug since it seems obsolete to center the window if it's being maximized anyway but it might still be nice to cover that case...
That case also causes weird behavior on macOS btw. where the window gets the right size (full screen size) but gets moved partially off screen.
